hermes-benchmark/src/jmh/java/pl/allegro/tech/hermes/benchmark/environment/BenchmarkProviders.java

```diff
@@ -0,0 +1,90 @@
+package pl.allegro.tech.hermes.benchmark.environment;
+
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.undertow.server.HttpHandler;
+import pl.allegro.tech.hermes.common.config.ConfigFactory;
+import pl.allegro.tech.hermes.common.message.wrapper.AvroMessageContentWrapper;
+import pl.allegro.tech.hermes.common.metric.HermesMetrics;
+import pl.allegro.tech.hermes.frontend.cache.topic.TopicsCache;
+import pl.allegro.tech.hermes.frontend.listeners.BrokerListeners;
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessageProducer;
+import pl.allegro.tech.hermes.frontend.publishing.handlers.HandlersChainFactory;
+import pl.allegro.tech.hermes.frontend.publishing.handlers.ThroughputLimiter;
+import pl.allegro.tech.hermes.frontend.publishing.handlers.end.MessageEndProcessor;
+import pl.allegro.tech.hermes.frontend.publishing.handlers.end.MessageErrorProcessor;
+import pl.allegro.tech.hermes.frontend.publishing.message.MessageContentTypeEnforcer;
+import pl.allegro.tech.hermes.frontend.publishing.message.MessageFactory;
+import pl.allegro.tech.hermes.frontend.publishing.metadata.DefaultHeadersPropagator;
+import pl.allegro.tech.hermes.frontend.server.HermesServer;
+import pl.allegro.tech.hermes.frontend.validator.MessageValidators;
+import pl.allegro.tech.hermes.metrics.PathsCompiler;
+import pl.allegro.tech.hermes.schema.DirectCompiledSchemaRepository;
+import pl.allegro.tech.hermes.schema.DirectSchemaVersionsRepository;
+import pl.allegro.tech.hermes.schema.RawSchemaClient;
+import pl.allegro.tech.hermes.schema.SchemaCompilersFactory;
+import pl.allegro.tech.hermes.schema.SchemaRepository;
+import pl.allegro.tech.hermes.test.helper.config.MutableConfigFactory;
+import pl.allegro.tech.hermes.tracker.frontend.Trackers;
+
+import java.io.IOException;
+import java.time.Clock;
+import java.util.Collections;
+
+import static pl.allegro.tech.hermes.api.TopicName.fromQualifiedName;
+import static pl.allegro.tech.hermes.benchmark.environment.HermesServerEnvironment.BENCHMARK_TOPIC;
+import static pl.allegro.tech.hermes.benchmark.environment.HermesServerEnvironment.loadMessageResource;
+import static pl.allegro.tech.hermes.common.config.Configs.FRONTEND_TOPIC_METADATA_REFRESH_JOB_ENABLED;
+import static pl.allegro.tech.hermes.frontend.publishing.handlers.ThroughputLimiter.QuotaInsight.quotaConfirmed;
+
+class BenchmarkProviders {
```

Comment:
After reading this name, I suspect that this class is responsible for providing benchmarks, which is not actually true as its purpose is to create a `HermesServer` instance for benchmarks. So, maybe it would be better to name it `HermesServerFactory`.